pygmt/figure.py

```diff
@@ -235,10 +235,14 @@ def psconvert(self, icc_gray=False, **kwargs):
                 kwargs["N"] = "+i"
             else:
                 kwargs["N"] += "+i"
-        # allow for spaces in figure name
-        kwargs["F"] = f'"{kwargs.get("F")}"' if kwargs.get("F") else None
+
+        # Manually handle prefix -F argument so spaces aren't converted to \040
+        # by build_arg_string function. For more information, see
+        # https://github.com/GenericMappingTools/pygmt/pull/1487
+        prefix = kwargs.pop("F")
```

Comment:
Should there be a check before this that ``prefix`` is provided so that the user doesn't get a KeyError here?

Reply:
Good point. Looking at the docs at https://docs.generic-mapping-tools.org/6.3/psconvert.html, it reads like `prefix`/`-F` is optional, but when I tried running `psconvert` without the `-F` option, it throws an error: `psconvert [ERROR]: Modern GMT mode requires the -F option`. See https://github.com/GenericMappingTools/gmt/blob/adb244afa51ca7246cc051080c9d47193087d6c2/src/psconvert.c#L1041-L1042

So since PyGMT is modern mode only, it should be fine to keep this line intact.

Reply:
The docs say "If no input files are given, will convert the current active figure (see pygmt.figure). In this case, an output name must be given using parameter prefix." How does one provide other input files than the current figure using pygmt.Figure.psconvert?

If we chose not to support operations analogous to ``psconvert -Tg test.ps`` (which uses the same prefix as the original file name), I think we should make ``prefix`` required in the function signature or raise an invalid input exception to avoid the traceback below. I don't think it's obvious to that the error is caused by not using ``prefix``.

<img width="980" alt="image" src="https://user-images.githubusercontent.com/14077947/158040519-f55e6f30-3e1d-43e0-83f2-bea6a536ffb7.png">

Reply:
> How does one provide other input files than the current figure using pygmt.Figure.psconvert?

We can't, because it has not been implemented yet. The current `fig.psconvert` is a method of the `pygmt.Figure()` class and is tied to the `fig`, and can only 'psconvert' the active figure.

> If we chose not to support operations analogous to `psconvert -Tg test.ps` (which uses the same prefix as the original file name), I think we should make `prefix` required in the function signature or raise an invalid input exception to avoid the traceback below. I don't think it's obvious to that the error is caused by not using `prefix`.

Ok, I've added a try-except block in 801ba01b3c83a134f6dd15f756223f8e3663513a to raise a GMTInvalidInput if `prefix`/`-F` is not given.